### PR TITLE
Terraform AWS DB module

### PR DESCRIPTION
**Pull request by klimantovich**
Module for terraform which creates RDS db

### CODE REVIEW

File: aws-rds/main.tf [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+# DB instance configuration
+#-----------------------------------------------
+resource "aws_db_instance" "default" {
+
+  allocated_storage      = var.replicate_source_db != null ? null : var.db_allocated_storage
+  apply_immediately      = var.db_apply_immediately
+  availability_zone      = var.db_multi_az ? null : var.db_availability_zone
+  ca_cert_identifier     = var.db_ca_cert_identifier
+  db_name                = var.database_name
+  db_subnet_group_name   = local.create_subnet_group ? join("", aws_db_subnet_group.default[*].name) : null
```
Comment (Jiraiya106) [inline aws-rds/main.tf:16]: What do this function?

File: aws-rds/main.tf [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  description              = "Ingress conn. form SGs"
+  security_group_id        = aws_security_group.db.id
+  from_port                = var.security_group_db_port
+  to_port                  = var.security_group_db_port
+  protocol                 = "tcp"
+  source_security_group_id = var.db_ingress_security_group_ids[count.index]
+}
+
+resource "aws_security_group_rule" "ingress_from_cidr_blocks" {
+  type              = "ingress"
```
Comment (Jiraiya106) [inline aws-rds/main.tf:59]: These things are done via dynamic block because a single group is rarely used

File: aws-rds/main.tf [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  description       = "Ingress conn. from IPs"
+  security_group_id = aws_security_group.db.id
+  from_port         = var.security_group_db_port
+  to_port           = var.security_group_db_port
+  protocol          = "tcp"
+  cidr_blocks       = var.db_ingress_cidr_blocks
+}
+
+resource "aws_security_group_rule" "egress" {
+  type              = "egress"
```
Comment (Jiraiya106) [inline aws-rds/main.tf:69]: These things are done via dynamic block because a single group is rarely used

File: aws-rds/variables.tf [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+}
+
+variable "db_multi_az" {
+  description = "Specifies if the RDS instance is multi-AZ"
+  type        = bool
+  default     = false
+}
+
+variable "db_password" {
+  description = "Required unless replicate_source_db is provided.) Password for the master DB user"
```
Comment (Jiraiya106) [inline aws-rds/variables.tf:82]: sensitive

Comment (klimantovich) [reply aws-rds/variables.tf:82]: done.
sensetive = true

File: aws-rds/main.tf (hunk 1 above)
Comment (klimantovich) [reply aws-rds/main.tf:16]: This function checks if variable "subnet_group_subnet_ids" is set, and than decide to create new subnet group or to use default.
As aws_db_subnet_group resource has "count" set, in this case function "join" transforms it to string type value

**Review by Jiraiya106 — APPROVED** [review]

**Pull request merged**